hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/DeleteBucketHandler.java

```diff
@@ -18,30 +18,164 @@
       throws IOException {
 
     String volumeName = address.getVolumeName();
     String bucketName = address.getBucketName();
-
     OzoneVolume vol = client.getObjectStore().getVolume(volumeName);
+
+    if (bRecursive) {
+      Collection<String> serviceIds = getConf().getTrimmedStringCollection(
+          OZONE_OM_SERVICE_IDS_KEY);
+      if (Strings.isNullOrEmpty(omServiceId)) {
+        if (serviceIds.size() > 1) {
+          out().printf("OmServiceID not provided, provide using " +
+              "-id <OM_SERVICE_ID>%n");
+          return;
+        } else if (serviceIds.size() == 1) {
+          // Only one OM service ID configured, we can use that
+          omServiceId = serviceIds.iterator().next();
+        }
+      }
+      if (!yes) {
+        // Ask for user confirmation
+        out().print("This command will delete bucket recursively." +
+            "\nThere is no recovery option after using this command, " +
+            "and no trash for FSO buckets." +
+            "\nEnter 'yes' to proceed': ");
+        out().flush();
+        Scanner scanner = new Scanner(new InputStreamReader(
+            System.in, StandardCharsets.UTF_8));
+        String confirmation = scanner.next().trim().toLowerCase();
+        if (!confirmation.equals("yes")) {
+          out().println("Operation cancelled.");
+          return;
+        }
+      }
+      OzoneBucket bucket = vol.getBucket(bucketName);
+      if (bucket.getBucketLayout().equals(OBJECT_STORE)) {
+        deleteOBSBucketRecursive(vol, bucket);
+      } else {
+        deleteFSBucketRecursive(vol, bucket);
+      }
+      return;
+    }
+    // Delete bucket without recursive
     vol.deleteBucket(bucketName);
+    out().printf("Bucket %s is deleted%n", bucketName);
   }
 
+  /**
+   * Delete OBS bucket recursively.
+   *
+   * @param bucket OzoneBucket
+   */
+  private void deleteOBSBucketRecursive(OzoneVolume vol, OzoneBucket bucket) {
+    ArrayList<String> keys = new ArrayList<>();
+    try {
+      if (!bucket.isLink()) {
+        Iterator<? extends OzoneKey> iterator = bucket.listKeys(null);
+        while (iterator.hasNext()) {
+          keys.add(iterator.next().getName());
+          if (MAX_KEY_DELETE_BATCH_SIZE == keys.size()) {
+            bucket.deleteKeys(keys);
+            keys.clear();
+          }
+        }
+        // delete if any remaining keys left
+        if (keys.size() > 0) {
+          bucket.deleteKeys(keys);
+        }
+      }
+      vol.deleteBucket(bucket.getName());
+      out().printf("Bucket %s is deleted%n", bucket.getName());
+    } catch (Exception e) {
+      out().println("Could not delete bucket.");
+    }
+  }
+
+  /**
+   * Delete Legacy/FSO bucket recursively.
+   *
+   * @param vol String
+   * @param bucket OzoneBucket
+   */
+  private void deleteFSBucketRecursive(OzoneVolume vol, OzoneBucket bucket) {
+    try {
+      String hostPrefix = OZONE_OFS_URI_SCHEME + "://";
+      if (!Strings.isNullOrEmpty(omServiceId)) {
```

Comment:
omServiceId is always nonNull/nonEmpty because of the earlier check on line 85. Checking it here again is not required.

Reply:
I think in case of non-HA, omServiceId will not be present. L85 checks if serviceIds are configured and not provided while executing command. If serviceIds are not configured then we are continuing it.

Reply:
I moved now no service ID configured case at L94, and removed checking here.